### PR TITLE
Added support for Macos on ARM. Updated all build scripts to libusb 1.0.24. Switched to libusb on GitHub.

### PR DESCRIPTION
**Pull request by git-moss**
Would be great if you could accept this for the official channel. Thanks!

### CODE REVIEW

**Comment by andriy-gerasika** [comment]
upvote (need libusb 1.0.23 and mac m1)

**Comment by llongi** [comment]
Please also update README.md and dist/bundle to mention the new darwin architecture in the right places.
After that looks fine to me, I'll try merging it and see if the CI servers still do something. :-)
Haven't been active on this for a long time and am not really using it much anymore.

**Comment by git-moss** [comment]
Thanks for looking into that! I updated the readme but I cannot update the bundle file since I have no access to your CI server.

File: dists/darwin/build [hunk 1]
```diff
@@ -12,60 +15,82 @@
...
+    # Unzip libusb
+    mkdir -p "$BUILD_DIR"
+    cd "$BUILD_DIR"
+    unzip "$LIBUSB_TARGET"
+    mv "$BUILD_DIR/libusb-$LIBUSB_VERSION" "$BUILD_DIR/libusb"
+    cd "$BUILD_DIR/libusb"
+
+    # Build libusb
+    autoreconf -i
+    ./configure --disable-shared --enable-static --with-pic --prefix="$ROOT_DIR" --host=$ARCH
```
Comment (hisano) [inline dists/darwin/build:71]: confiture's `host` option should contain `-darwin`.
No `-darwin` choose the null backend of libusb. As a result, usb4java returns no device.
https://github.com/libusb/libusb/blob/v1.0.26/configure.ac#L76

**Pull request merged**